vita3k/renderer/src/texture_cache.cpp

```diff
@@ -295,6 +296,29 @@ static void convert_x8u24_to_f32(void *dest, const void *data, const uint32_t wi
     }
 }
 
+static void convert_U8U3U3U2_to_U8U8U8U8(void *dest, const void *data, const uint32_t width, const uint32_t height, const uint32_t pixels_per_stride) {
+    auto dst = static_cast<uint32_t *>(dest);
+    auto src = static_cast<const uint16_t *>(data);
+
+    for (uint32_t row = 0; row < height; ++row) {
+        for (uint32_t col = 0; col < width; ++col) {
+            const uint32_t src_value = src[col];
+            uint32_t c1 = (src_value & 0xFF00) >> 8;
+            uint32_t c2 = (src_value & 0x00E0) >> 5;
+            uint32_t c3 = (src_value & 0x001C) >> 2;
+            uint32_t c4 = (src_value & 0x0003);
+
+            uint32_t value = (c1 << 24);
+            value |= ((c2 && 0x4) << 23) | ((c2 && 0x4) << 22) | ((c2 && 0x4) << 21) | ((c2 && 0x02) << 20) | ((c2 && 0x02) << 19) | ((c2 && 0x02) << 18) | ((c2 && 0x01) << 17) | ((c2 && 0x01) << 16);
+            value |= ((c3 && 0x4) << 15) | ((c3 && 0x4) << 14) | ((c3 && 0x4) << 13) | ((c3 && 0x02) << 12) | ((c3 && 0x02) << 11) | ((c3 && 0x02) << 10) | ((c3 && 0x01) << 9) | ((c3 && 0x01) << 8);
+            value |= ((c4 && 0x2) << 7) | ((c4 && 0x2) << 6) | ((c4 && 0x2) << 5) | ((c4 && 0x02) << 4) | ((c4 && 0x01) << 3) | ((c4 && 0x01) << 2) | ((c4 && 0x01) << 1) | ((c4 && 0x01));
```

Comment:
That's completely wrong, stop trying to use ChatGPT to generate this kind of code

Reply:
Euu ? Is not gpt att all 🗿